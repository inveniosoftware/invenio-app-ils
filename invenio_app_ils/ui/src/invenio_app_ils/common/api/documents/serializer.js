import isEmpty from 'lodash/isEmpty';
import { fromISO } from '../date';

function serializeSeries(series) {
  return {
    serial: series.filter(s => s.mode_of_issuance === 'SERIAL'),
    multipart: series.filter(s => s.mode_of_issuance === 'MULTIPART_MONOGRAPH'),
  };
}

function serializeResponse(hit) {
  let result = {};
  if (!isEmpty(hit)) {
    result['id'] = hit.id;
    result['created'] = fromISO(hit.created);
    result['updated'] = fromISO(hit.updated);
    if (hit.links) {
      result['links'] = hit.links;
    }
    if (!isEmpty(hit.metadata)) {
      result['metadata'] = hit.metadata;
      if (!isEmpty(hit.metadata.circulation.next_available_date)) {
        result['metadata']['circulation']['next_available_date'] = new Date(
          result['metadata']['circulation']['next_available_date']
        ).toDateString();
      }
      if (!isEmpty(hit.metadata.series)) {
        // Split the series based on mode of issuance - requested by the library
        result['metadata']['series'] = serializeSeries(hit.metadata.series);
      }
      result['pid'] = hit.metadata.pid;
    }
  }
  return result;
}

export const serializer = {
  fromJSON: serializeResponse,
};
