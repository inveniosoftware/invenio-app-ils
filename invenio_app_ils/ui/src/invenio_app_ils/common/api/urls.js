export const ApiURLS = {
  loans: {
    loan: '/circulation/loans/:loanPid',
    list: '/circulation/loans/',
    checkout: '/circulation/loans/checkout',
    request: '/circulation/loans/request',
  },
  items: {},
  documents: {},
  locations: {},
  users: {},
};
