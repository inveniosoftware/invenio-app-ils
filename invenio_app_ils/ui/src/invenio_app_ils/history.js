import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

export const goToHandler = path => e => {
  history.push(path);
};

export const goTo = (path, state = {}) => {
  history.push(path, state);
};

export default history;
