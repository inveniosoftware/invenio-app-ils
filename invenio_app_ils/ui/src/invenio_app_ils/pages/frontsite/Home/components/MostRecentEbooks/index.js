import { connect } from 'react-redux';
import { fetchMostRecentEbooks } from './state/actions';
import MostRecentEbooksComponent from './MostRecentEbooks';

const mapStateToProps = state => ({
  data: state.mostRecentEbooks.data,
  error: state.mostRecentEbooks.error,
  isLoading: state.mostRecentEbooks.isLoading,
  hasError: state.mostRecentEbooks.hasError,
});

const mapDispatchToProps = dispatch => ({
  fetchMostRecentEbooks: () => dispatch(fetchMostRecentEbooks()),
});

export const MostRecentEbooks = connect(
  mapStateToProps,
  mapDispatchToProps
)(MostRecentEbooksComponent);
