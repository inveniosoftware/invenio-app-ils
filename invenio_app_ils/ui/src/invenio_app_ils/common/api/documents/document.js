import { http } from '../base';
import { serializer } from './serializer';

const documentURL = '/documents/';

const get = documentPid => {
  return http.get(`${documentURL}${documentPid}`).then(response => {
    response.data = serializer.fromJSON(response.data);
    return response;
  });
};

class QueryBuilder {
  constructor() {
    this.overbookedQuery = [];
    this.availableItemsQuery = [];
    this.withPendingLoansQuery = [];
  }

  overbooked() {
    this.overbookedQuery.push(`circulation.overbooked:true`);
    return this;
  }

  withAvailableItems() {
    this.availableItemsQuery.push('circulation.loanable_items:>0');
    return this;
  }

  withPendingLoans() {
    this.withPendingLoansQuery.push('circulation.pending_loans:>0');
    return this;
  }

  qs() {
    return this.overbookedQuery
      .concat(this.availableItemsQuery, this.withPendingLoansQuery)
      .join(' AND ');
  }
}

const queryBuilder = () => {
  return new QueryBuilder();
};

const list = query => {
  return http.get(`${documentURL}?q=${query}`).then(response => {
    const totalHits = response.data.hits.total;
    response.data = response.data.hits.hits.map(hit =>
      serializer.fromJSON(hit)
    );
    response.data.totalHits = totalHits;
    return response;
  });
};

const count = query => {
  return http.get(`${documentURL}?q=${query}`).then(response => {
    response.data = response.data.hits.total;
    return response;
  });
};

export const document = {
  get: get,
  list: list,
  count: count,
  query: queryBuilder,
  serializer: serializer,
  url: documentURL,
};
