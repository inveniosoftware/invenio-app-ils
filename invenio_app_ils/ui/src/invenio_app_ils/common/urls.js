import { generatePath } from 'react-router-dom';
import { loan as loanApi } from './api/loan';
import { item as itemApi } from './api/item';

export const FrontSiteURLS = {};

const BackOfficeBase = '/backoffice';

export const BackOfficeURLS = {
  home: BackOfficeBase,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  itemsSearch: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansSearch: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronDetails: `${BackOfficeBase}/users/:userPid`,
};

export const viewLoanDetailsUrl = loanPid => {
  return generatePath(BackOfficeURLS.loanDetails, {
    loanPid: loanPid,
  });
};

export const showAllLoansUrl = (
  documentPid,
  itemPid,
  state,
  patronPid,
  extraQuery
) => {
  const qs = loanApi.buildLoansQuery(
    documentPid,
    itemPid,
    state,
    patronPid,
    extraQuery
  );
  return `${BackOfficeURLS.loansSearch}?q=${qs}`;
};

export const viewItemDetailsUrl = itemPid => {
  return generatePath(BackOfficeURLS.itemDetails, {
    itemPid: itemPid,
  });
};

export const showAllItemsUrl = (documentPid, state, extraQuery) => {
  const qs = itemApi.buildItemsQuery(documentPid, state, extraQuery);
  return `${BackOfficeURLS.itemsSearch}?q=${qs}`;
};
