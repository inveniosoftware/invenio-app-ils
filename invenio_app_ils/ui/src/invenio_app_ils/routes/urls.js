import { generatePath } from 'react-router-dom';
import { invenioConfig } from '../common/config';

const FrontSiteBase = '/';

const FrontSiteRoutesList = {
  home: FrontSiteBase,
  patronProfile: `${FrontSiteBase}profile`,
  documentsList: `${FrontSiteBase}search`,
  documentDetails: `${FrontSiteBase}records/:documentPid`,
};

const FrontSiteRoutesGenerators = {
  documentsListWithQuery: qs => `${FrontSiteRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(FrontSiteRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
};

export const FrontSiteRoutes = {
  ...FrontSiteRoutesGenerators,
  ...FrontSiteRoutesList,
};

const BackOfficeBase = '/backoffice';

const BackOfficeRoutesList = {
  home: BackOfficeBase,
  documentsList: `${BackOfficeBase}/documents`,
  documentsCreate: `${BackOfficeBase}/documents/create`,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  eitemsList: `${BackOfficeBase}/eitems`,
  eitemDetails: `${BackOfficeBase}/eitems/:eitemPid`,
  itemsList: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansList: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronsList: `${BackOfficeBase}/patrons`,
  patronDetails: `${BackOfficeBase}/patrons/:patronPid`,
  locationsList: `${BackOfficeBase}/locations`,
  seriesList: `${BackOfficeBase}/series`,
  seriesDetails: `${BackOfficeBase}/series/:seriesPid`,
};

const BackOfficeRouteGenerators = {
  documentsListWithQuery: qs => `${BackOfficeRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(BackOfficeRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
  eitemDetailsFor: eitemPid =>
    generatePath(BackOfficeRoutesList.eitemDetails, { eitemPid: eitemPid }),
  itemsListWithQuery: qs => `${BackOfficeRoutesList.itemsList}?q=${qs}`,
  itemDetailsFor: itemPid =>
    generatePath(BackOfficeRoutesList.itemDetails, { itemPid: itemPid }),
  loansListWithQuery: qs => `${BackOfficeRoutesList.loansList}?q=${qs}`,
  loanDetailsFor: loanPid =>
    generatePath(BackOfficeRoutesList.loanDetails, { loanPid: loanPid }),
  patronDetailsFor: patronPid =>
    generatePath(BackOfficeRoutesList.patronDetails, { patronPid: patronPid }),
  seriesListWithQuery: qs => `${BackOfficeRoutesList.seriesList}?q=${qs}`,
  seriesDetailsFor: seriesPid =>
    generatePath(BackOfficeRoutesList.seriesDetails, { seriesPid: seriesPid }),
};

export const BackOfficeRoutes = {
  ...BackOfficeRoutesList,
  ...BackOfficeRouteGenerators,
};

export const openRecordEditor = (path, recid = '') => {
  window.open(`${invenioConfig.editor.url}${path}${recid}`, '_blank');
};
