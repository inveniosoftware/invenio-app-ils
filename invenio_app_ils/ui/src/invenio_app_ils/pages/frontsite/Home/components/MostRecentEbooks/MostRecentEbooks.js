import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Button, Card, Divider } from 'semantic-ui-react';
import { Error, Loader } from '../../../../../common/components';
import { goToHandler } from '../../../../../history';
import { FrontSiteRoutes } from '../../../../../routes/urls';
import { EbookCard } from '../EbookCard';

export default class MostRecentEbooks extends Component {
  constructor(props) {
    super(props);
    this.fetchMostRecentEbooks = props.fetchMostRecentEbooks;
    this.maxItemsToDisplay = props.maxDisplayedItems;
  }

  componentDidMount() {
    this.fetchMostRecentEbooks();
  }

  prepareData(data) {
    const ebooks = data.hits.slice(0, this.maxItemsToDisplay).map(book => {
      return book.metadata.eitems.hits.map(ebook => {
        return {
          document_pid: book.metadata.pid,
          pid: ebook.pid,
          title: book.metadata.title.title,
          description: ebook.description,
          open_access: ebook.open_access,
          imageSize: 'small',
          imageCover: 'https://assets.thalia.media/img/46276899-00-00.jpg',
          onClick: goToHandler(
            FrontSiteRoutes.documentDetailsFor(book.metadata.pid)
          ),
        };
      });
    });
    return (
      []
        .concat(...ebooks)
        // slice again because documents could contain multiple eitems
        .slice(0, this.maxItemsToDisplay)
    );
  }

  renderCards(ebooks) {
    const cards = ebooks.map(ebook => {
      return <EbookCard key={ebook.pid} ebookData={ebook} />;
    });

    return <Card.Group>{cards}</Card.Group>;
  }

  render() {
    const { data, isLoading, error } = this.props;
    const ebooks = this.prepareData(data);
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>
          <div>
            <Button
              content="See All"
              onClick={goToHandler(
                FrontSiteRoutes.documentsListWithQuery(
                  '&sort=mostrecent&order=desc' +
                    '&aggr[0][has_eitems][value]=has_eitems.electronic versions'
                )
              )}
            />
            <Divider hidden />
            {this.renderCards(ebooks)}
          </div>
        </Error>
      </Loader>
    );
  }
}

MostRecentEbooks.propTypes = {
  fetchMostRecentEbooks: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired,
};
