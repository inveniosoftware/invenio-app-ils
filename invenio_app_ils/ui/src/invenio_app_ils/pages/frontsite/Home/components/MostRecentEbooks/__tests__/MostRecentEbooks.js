import React from 'react';
import { shallow, mount } from 'enzyme';
import { Settings } from 'luxon';
import { FrontSiteRoutes } from '../../../../../../routes/urls';
import MostRecentEbooks from '../MostRecentEbooks';
import history from '../../../../../../history';
import * as testData from '../../../../../../../../../../tests/data/documents.json';

Settings.defaultZoneName = 'utc';

describe('MostRecentEbooks tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const data = {
    hits: [
      {
        metadata: {
          ...testData[0],
          pid: '123',
          eitems: {
            total: 1,
            hits: [
              {
                document_pid: 'doc1',
                pid: 'eitem1',
                title: 'title1',
                description: 'Lorem ipsum',
                open_access: true,
                image_size: '',
                image_cover: '',
                onClick: jest.fn(),
              },
            ],
          },
        },
      },
      {
        metadata: {
          ...testData[1],
          pid: '456',
          eitems: {
            total: 1,
            hits: [
              {
                document_pid: 'doc2',
                pid: 'eitem2',
                title: 'title2',
                description: 'Lorem ipsum',
                open_access: false,
                image_size: '',
                image_cover: '',
                onClick: jest.fn(),
              },
            ],
          },
        },
      },
    ],
    total: 2,
  };

  it('should load the most recent ebooks component', () => {
    const component = shallow(
      <MostRecentEbooks
        data={{ hits: [], total: 0 }}
        fetchMostRecentEbooks={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch most recent ebooks on mount', () => {
    const mockedFetchMostRecentEbooks = jest.fn();
    component = mount(
      <MostRecentEbooks
        data={{ hits: [], total: 0 }}
        fetchMostRecentEbooks={mockedFetchMostRecentEbooks}
      />
    );
    expect(mockedFetchMostRecentEbooks).toHaveBeenCalled();
  });

  it('should render list of most recent ebooks', () => {
    component = mount(
      <MostRecentEbooks data={data} fetchMostRecentEbooks={() => {}} />
    );

    expect(component).toMatchSnapshot();
    const cards = component
      .find('MostRecentEbooks')
      .find('CardGroup')
      .find('EbookCard')
      .find('Card')
      .filterWhere(
        element =>
          element.prop('data-test') === 'eitem1' ||
          element.prop('data-test') === 'eitem2'
      );
    expect(cards).toHaveLength(2);
  });

  it('should render the see all button when showing most recent ebooks', () => {
    component = mount(
      <MostRecentEbooks
        data={data}
        fetchMostRecentEbooks={() => {}}
        maxItemsToDisplay={1}
      />
    );

    expect(component).toMatchSnapshot();
    const button = component
      .find('MostRecentEbooks')
      .find('Button')
      .filterWhere(element => element.prop('content') === 'See All');
    expect(button).toHaveLength(1);
  });

  it('should go to book details when clicking on a book', () => {
    const mockedHistoryPush = jest.fn();
    history.push = mockedHistoryPush;

    component = mount(
      <MostRecentEbooks
        data={data}
        fetchMostRecentEbooks={() => {}}
        maxItemsToDisplay={1}
      />
    );
    expect(component).toMatchSnapshot();

    const docPid = data.hits[0].metadata.pid;
    const button = component
      .find('MostRecentEbooks')
      .find('CardGroup')
      .find('EbookCard')
      .find('Card')
      .filterWhere(element => element.prop('data-test') === 'eitem1')
      .find('a');
    button.simulate('click');

    const expectedParam = FrontSiteRoutes.documentDetailsFor(docPid);
    expect(mockedHistoryPush).toHaveBeenCalledWith(expectedParam);
  });
});
