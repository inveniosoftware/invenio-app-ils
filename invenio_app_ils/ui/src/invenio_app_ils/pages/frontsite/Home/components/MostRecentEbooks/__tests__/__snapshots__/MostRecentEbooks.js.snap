// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostRecentEbooks tests should go to book details when clicking on a book 1`] = `
<MostRecentEbooks
  data={
    Object {
      "hits": Array [
        Object {
          "metadata": Object {
            "abstracts": Array [
              Object {
                "value": "This is an abstract",
              },
            ],
            "authors": Array [
              Object {
                "full_name": "Jack E. Davis",
              },
            ],
            "edition": "ed. 1",
            "eitems": Object {
              "hits": Array [
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "doc1",
                  "image_cover": "",
                  "image_size": "",
                  "onClick": [MockFunction],
                  "open_access": true,
                  "pid": "eitem1",
                  "title": "title1",
                },
              ],
              "total": 1,
            },
            "language": Array [
              "en",
            ],
            "pid": "123",
            "title": Object {
              "title": "The Gulf: The Making of An American Sea",
            },
          },
        },
        Object {
          "metadata": Object {
            "abstracts": Array [
              Object {
                "value": "This is an abstract",
              },
            ],
            "authors": Array [
              Object {
                "full_name": "Caroline Fraser",
              },
            ],
            "eitems": Object {
              "hits": Array [
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "doc2",
                  "image_cover": "",
                  "image_size": "",
                  "onClick": [MockFunction],
                  "open_access": false,
                  "pid": "eitem2",
                  "title": "title2",
                },
              ],
              "total": 1,
            },
            "language": Array [
              "it",
            ],
            "pid": "456",
            "title": Object {
              "title": "Prairie Fires: The American Dreams of Laura Ingalls Wilder",
            },
          },
        },
      ],
      "total": 2,
    }
  }
  fetchMostRecentEbooks={[Function]}
  maxItemsToDisplay={1}
>
  <Loader>
    <Error>
      <div>
        <Button
          as="button"
          content="See All"
          onClick={[Function]}
        >
          <button
            className="ui button"
            onClick={[Function]}
          >
            See All
          </button>
        </Button>
        <Divider
          hidden={true}
        >
          <div
            className="ui hidden divider"
          />
        </Divider>
        <CardGroup>
          <div
            className="ui cards"
          >
            <EbookCard
              ebookData={
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "123",
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "open_access": true,
                  "pid": "eitem1",
                  "title": "The Gulf: The Making of An American Sea",
                }
              }
              key="eitem1"
            >
              <Card
                data-test="eitem1"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="eitem1"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          The Gulf: The Making of An American Sea
                        </div>
                      </CardHeader>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </EbookCard>
            <EbookCard
              ebookData={
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "456",
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "open_access": false,
                  "pid": "eitem2",
                  "title": "Prairie Fires: The American Dreams of Laura Ingalls Wilder",
                }
              }
              key="eitem2"
            >
              <Card
                data-test="eitem2"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="eitem2"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          Prairie Fires: The American Dreams of Laura Ingalls Wilder
                        </div>
                      </CardHeader>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </EbookCard>
          </div>
        </CardGroup>
      </div>
    </Error>
  </Loader>
</MostRecentEbooks>
`;

exports[`MostRecentEbooks tests should load the most recent ebooks component 1`] = `
<Loader>
  <Error>
    <div>
      <Button
        as="button"
        content="See All"
        onClick={[Function]}
      />
      <Divider
        hidden={true}
      />
      <CardGroup />
    </div>
  </Error>
</Loader>
`;

exports[`MostRecentEbooks tests should render list of most recent ebooks 1`] = `
<MostRecentEbooks
  data={
    Object {
      "hits": Array [
        Object {
          "metadata": Object {
            "abstracts": Array [
              Object {
                "value": "This is an abstract",
              },
            ],
            "authors": Array [
              Object {
                "full_name": "Jack E. Davis",
              },
            ],
            "edition": "ed. 1",
            "eitems": Object {
              "hits": Array [
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "doc1",
                  "image_cover": "",
                  "image_size": "",
                  "onClick": [MockFunction],
                  "open_access": true,
                  "pid": "eitem1",
                  "title": "title1",
                },
              ],
              "total": 1,
            },
            "language": Array [
              "en",
            ],
            "pid": "123",
            "title": Object {
              "title": "The Gulf: The Making of An American Sea",
            },
          },
        },
        Object {
          "metadata": Object {
            "abstracts": Array [
              Object {
                "value": "This is an abstract",
              },
            ],
            "authors": Array [
              Object {
                "full_name": "Caroline Fraser",
              },
            ],
            "eitems": Object {
              "hits": Array [
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "doc2",
                  "image_cover": "",
                  "image_size": "",
                  "onClick": [MockFunction],
                  "open_access": false,
                  "pid": "eitem2",
                  "title": "title2",
                },
              ],
              "total": 1,
            },
            "language": Array [
              "it",
            ],
            "pid": "456",
            "title": Object {
              "title": "Prairie Fires: The American Dreams of Laura Ingalls Wilder",
            },
          },
        },
      ],
      "total": 2,
    }
  }
  fetchMostRecentEbooks={[Function]}
>
  <Loader>
    <Error>
      <div>
        <Button
          as="button"
          content="See All"
          onClick={[Function]}
        >
          <button
            className="ui button"
            onClick={[Function]}
          >
            See All
          </button>
        </Button>
        <Divider
          hidden={true}
        >
          <div
            className="ui hidden divider"
          />
        </Divider>
        <CardGroup>
          <div
            className="ui cards"
          >
            <EbookCard
              ebookData={
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "123",
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "open_access": true,
                  "pid": "eitem1",
                  "title": "The Gulf: The Making of An American Sea",
                }
              }
              key="eitem1"
            >
              <Card
                data-test="eitem1"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="eitem1"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          The Gulf: The Making of An American Sea
                        </div>
                      </CardHeader>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </EbookCard>
            <EbookCard
              ebookData={
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "456",
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "open_access": false,
                  "pid": "eitem2",
                  "title": "Prairie Fires: The American Dreams of Laura Ingalls Wilder",
                }
              }
              key="eitem2"
            >
              <Card
                data-test="eitem2"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="eitem2"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          Prairie Fires: The American Dreams of Laura Ingalls Wilder
                        </div>
                      </CardHeader>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </EbookCard>
          </div>
        </CardGroup>
      </div>
    </Error>
  </Loader>
</MostRecentEbooks>
`;

exports[`MostRecentEbooks tests should render the see all button when showing most recent ebooks 1`] = `
<MostRecentEbooks
  data={
    Object {
      "hits": Array [
        Object {
          "metadata": Object {
            "abstracts": Array [
              Object {
                "value": "This is an abstract",
              },
            ],
            "authors": Array [
              Object {
                "full_name": "Jack E. Davis",
              },
            ],
            "edition": "ed. 1",
            "eitems": Object {
              "hits": Array [
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "doc1",
                  "image_cover": "",
                  "image_size": "",
                  "onClick": [MockFunction],
                  "open_access": true,
                  "pid": "eitem1",
                  "title": "title1",
                },
              ],
              "total": 1,
            },
            "language": Array [
              "en",
            ],
            "pid": "123",
            "title": Object {
              "title": "The Gulf: The Making of An American Sea",
            },
          },
        },
        Object {
          "metadata": Object {
            "abstracts": Array [
              Object {
                "value": "This is an abstract",
              },
            ],
            "authors": Array [
              Object {
                "full_name": "Caroline Fraser",
              },
            ],
            "eitems": Object {
              "hits": Array [
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "doc2",
                  "image_cover": "",
                  "image_size": "",
                  "onClick": [MockFunction],
                  "open_access": false,
                  "pid": "eitem2",
                  "title": "title2",
                },
              ],
              "total": 1,
            },
            "language": Array [
              "it",
            ],
            "pid": "456",
            "title": Object {
              "title": "Prairie Fires: The American Dreams of Laura Ingalls Wilder",
            },
          },
        },
      ],
      "total": 2,
    }
  }
  fetchMostRecentEbooks={[Function]}
  maxItemsToDisplay={1}
>
  <Loader>
    <Error>
      <div>
        <Button
          as="button"
          content="See All"
          onClick={[Function]}
        >
          <button
            className="ui button"
            onClick={[Function]}
          >
            See All
          </button>
        </Button>
        <Divider
          hidden={true}
        >
          <div
            className="ui hidden divider"
          />
        </Divider>
        <CardGroup>
          <div
            className="ui cards"
          >
            <EbookCard
              ebookData={
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "123",
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "open_access": true,
                  "pid": "eitem1",
                  "title": "The Gulf: The Making of An American Sea",
                }
              }
              key="eitem1"
            >
              <Card
                data-test="eitem1"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="eitem1"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          The Gulf: The Making of An American Sea
                        </div>
                      </CardHeader>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </EbookCard>
            <EbookCard
              ebookData={
                Object {
                  "description": "Lorem ipsum",
                  "document_pid": "456",
                  "imageCover": "https://assets.thalia.media/img/46276899-00-00.jpg",
                  "imageSize": "small",
                  "onClick": [Function],
                  "open_access": false,
                  "pid": "eitem2",
                  "title": "Prairie Fires: The American Dreams of Laura Ingalls Wilder",
                }
              }
              key="eitem2"
            >
              <Card
                data-test="eitem2"
                onClick={[Function]}
              >
                <a
                  className="ui card"
                  data-test="eitem2"
                  onClick={[Function]}
                >
                  <Image
                    as="img"
                    size="small"
                    src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    ui={true}
                  >
                    <img
                      className="ui small image"
                      src="https://assets.thalia.media/img/46276899-00-00.jpg"
                    />
                  </Image>
                  <CardContent>
                    <div
                      className="content"
                    >
                      <CardHeader>
                        <div
                          className="header"
                        >
                          Prairie Fires: The American Dreams of Laura Ingalls Wilder
                        </div>
                      </CardHeader>
                    </div>
                  </CardContent>
                </a>
              </Card>
            </EbookCard>
          </div>
        </CardGroup>
      </div>
    </Error>
  </Loader>
</MostRecentEbooks>
`;
